File: patch_modal.py
import sys

path = 'frontend/src/app/admin/page.tsx'
content = open(path, 'r', encoding='utf-8').read()

marker = '      {/* MODAL DE GESTIÓN DE DOCUMENTOS DE CONTRATO */}'
start = content.index(marker)
end = content.index('      )}', start) + len('      )}')

replacement = (
    '      {/* MODAL DE GESTIÓN DE DOCUMENTOS DE CONTRATO */}\n'
    '      {isDocsModalOpen && selectedContractForDocs && (\n'
    '        <ContractAuditModal\n'
    "          isOpen={isDocsModalOpen}\n"
    "          onClose={() => setIsDocsModalOpen(false)}\n"
    "          contract={selectedContractForDocs}\n"
    "          token={getToken() || ''}\n"
    "          contractDocuments={contractDocuments}\n"
    "          setContractDocuments={setContractDocuments}\n"
    '        />\n'
    '      )}'
)

# Also add import at top if missing
import_line = "import { ContractAuditModal } from '@/components/modules/contracts/ContractAuditModal';\n"
if import_line.strip() not in content:
    drop_filter_import = "import { DropdownFilter } from '@/components/ui/DropdownFilter';"
    content = content.replace(drop_filter_import, drop_filter_import + '\n' + import_line.rstrip())

new_content = content[:start] + replacement + content[end:]
open(path, 'w', encoding='utf-8').write(new_content)
print('Done. Replaced', end - start, 'chars starting at', start)

File: patch_modal.js
const fs = require('fs');
const path = 'frontend/src/app/admin/page.tsx';
let content = fs.readFileSync(path, 'utf-8');

// 1. Add import if missing
const importLine = `import { ContractAuditModal } from '@/components/modules/contracts/ContractAuditModal';`;
if (!content.includes(importLine)) {
  content = content.replace(
    `import { DropdownFilter } from '@/components/ui/DropdownFilter';`,
    `import { DropdownFilter } from '@/components/ui/DropdownFilter';\n${importLine}`
  );
}

// 2. Replace the docs modal block
const modalStart = `      {/* MODAL DE GESTIÓN DE DOCUMENTOS DE CONTRATO */}`;
const modalEnd = `      )}`;

const startIdx = content.indexOf(modalStart);
if (startIdx === -1) { console.error('Modal start not found'); process.exit(1); }

// Find the closing )} AFTER the modal start
let searchFrom = startIdx + modalStart.length;
const endIdx = content.indexOf(modalEnd, searchFrom) + modalEnd.length;

const replacement = `      {/* MODAL DE GESTIÓN DE DOCUMENTOS DE CONTRATO */}
      {isDocsModalOpen && selectedContractForDocs && (
        <ContractAuditModal
          isOpen={isDocsModalOpen}
          onClose={() => setIsDocsModalOpen(false)}
          contract={selectedContractForDocs}
          token={getToken() || ''}
          contractDocuments={contractDocuments}
          setContractDocuments={setContractDocuments}
        />
      )}`;

content = content.slice(0, startIdx) + replacement + content.slice(endIdx);
fs.writeFileSync(path, content, 'utf-8');
console.log('Done. Modal replaced successfully.');
